lib/dl_core/dl_core/connection_executors/remote_query_executor/app_async.py

```diff
@@ -250,15 +251,19 @@ async def post(self) -> Union[web.Response, web.StreamResponse]:
                         ipaddress.ip_address(target_host)
                         host = target_host
                     except ValueError:
-                        try:
-                            resolver = aiodns.DNSResolver()
-                            resp = await resolver.query(target_host, "A")
-                            host = resp[0].host
-                        except aiodns.error.DNSError:
+                        resolver = aiodns.DNSResolver()
+                        for query_type in ("A", "AAAA"):
+                            try:
+                                resp = await resolver.query(target_host, query_type)
```

Comment:
I think we can do it in one call: `resolver.gethostbyname(host, socket.AF_UNSPEC)` (requires some parsing), but I haven't checked it well